src/main/java/com/autotune/analyzer/utils/EvalExParser.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    /**
+     * validate the objective function and return boolean based on the result
+     * @param objFunction
+     * @param functionVariables
+     * @return
+     */
+    @Override
+    public Boolean validate(String objFunction, ArrayList<Metric> functionVariables) {
+
+        /*
+            Remove all the variables from the objFunction matching with variables present in variableNames List
+            so that the function is left with only brackets and operators
+         */
+        String objFunctionOperators = "";
+        objFunctionOperators = objFunction.replaceAll("[a-zA-Z _]", "");
+
+        LOGGER.info("Objective Func Operators: {}", objFunctionOperators);
+
+        //Remove extra whitespaces from the string
+        objFunctionOperators = objFunctionOperators.replaceAll("\\s", "");
+
+        //check if there's any unnecessary character apart from the mathematical operators
+        if (!containsValidMathOperators(objFunctionOperators)) {
+            return false;
+        }
+
+        // check if brackets in objective function string are balanced
+        return areBracketsBalanced(objFunctionOperators);
+    }
+
+    /**
+     * check if the objFunction received contains valid mathematical operators
+     * @param objFunctionOperators
+     * @return
+     */
+    private boolean containsValidMathOperators(String objFunctionOperators) {
+
+        Deque<Character> mathOperators = new ArrayDeque<>();
+        for (int i = 0; i < objFunctionOperators.length(); i++) {
+            char charInObjFunction = objFunctionOperators.charAt(i);
+            if (!(charInObjFunction == ')'
+                    || charInObjFunction == '('
+                    || isNumeric(String.valueOf(charInObjFunction))
+                    || charInObjFunction == '.'
+            )) {
+                mathOperators.push(charInObjFunction);
+            }
+        }
+        for (Character mathOperator : mathOperators) {
+
+            if (!AutotuneSupportedTypes.MATH_OPERATORS_SUPPORTED.contains(String.valueOf(mathOperator))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * check if the string contains only numbers and return boolean result
+     * @param strNum
+     * @return
+     */
+    private static boolean isNumeric(String strNum) {
+
+        if (strNum == null) {
+            return false;
+        }
+        try {
+            Double.parseDouble(strNum);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * check if the brackets present in the objective function are balanced
+     * @param objFunction
+     * @return
+     */
+    private boolean areBracketsBalanced(String objFunction) {
+
+        Deque<Character> stack = new ArrayDeque<>();
+
+        for (int i = 0; i < objFunction.length(); i++) {
+            char bracket = objFunction.charAt(i);
+            if (bracket == ')') {
```

Comment:
What happens if brackets other than `()` used?

Reply:
This will be checked in this method only and will return false, if any other bracket or special character is present.

`private boolean containsValidMathOperators(String objFunctionOperators) {

        Deque<Character> mathOperators = new ArrayDeque<>();
        for (int i = 0; i < objFunctionOperators.length(); i++) {
            char charInObjFunction = objFunctionOperators.charAt(i);
            if (!(charInObjFunction == ')'
                    || charInObjFunction == '('
                    || isNumeric(String.valueOf(charInObjFunction))
                    || charInObjFunction == '.'
            )) {
                mathOperators.push(charInObjFunction);
            }
        }
        for (Character mathOperator : mathOperators) {

            if (!AutotuneSupportedTypes.MATH_OPERATORS_SUPPORTED.contains(String.valueOf(mathOperator))) {
                return false;
            }
        }
        return true;
    }`